src/blockchain/staking/stake.ts

```diff
@@ -0,0 +1,11 @@
+import { BaseProvider } from '@ethersproject/providers'
+import { BigNumber } from 'ethers'
+
+import { getXtkStakingContract } from './helper'
+
+export const stakeXtk = async (provider: BaseProvider, amount: BigNumber) => {
+  if (amount.lte(BigNumber.from(0))) return
```

Comment:
Missing error message!
```suggestion
  if (amount.lte(BigNumber.from(0))) {
    return Promise.reject(new Error('Invalid value for amount'))
  }
```